### ISSUE TITLE
[Crash] using structures as doors

### ISSUE THREAD

**Issue by DuncanLevings** [CreativeMD/LittleTiles#225]
Forge: 14.23.5.2838
CreativeCore: 1.9.61
LittleTiles: 1.5.0-pre157

Added upon latest Stoneblock 2 pack.

Description:
Creating a custom door with any animation (I tested with sliding and advanced door using child door), eventually crashed the game after repeated opening and closing.

> Description: Unexpected error
> 
> java.lang.StackOverflowError
>     at com.creativemd.littletiles.common.structure.LittleStructure.hasLoaded(LittleStructure.java:342)
>     at com.creativemd.littletiles.common.structure.LittleStructure.loadChildren(LittleStructure.java:441)
>     at com.creativemd.littletiles.common.structure.LittleStructure.loadChildren(LittleStructure.java:441)
>     at com.creativemd.littletiles.common.structure.LittleStructure.loadChildren(LittleStructure.java:441)
>     at com.creativemd.littletiles.common.structure.LittleStructure.loadChildren(LittleStructure.java:441)
>     at com.creativemd.littletiles.common.structure.LittleStructure.loadChildren(LittleStructure.java:441)
>     at com.creativemd.littletiles.common.structure.LittleStructure.loadChildren(LittleStructure.java:441)
> and so on...

Crash log:
[crash-2019-08-06_21.31.32-client.txt](https://github.com/CreativeMD/LittleTiles/files/3474928/crash-2019-08-06_21.31.32-client.txt)

**Comment by CreativeMD** [comment]
Can you reproduce this? This is a known issue ... but I could never fix it because it seems to happen very very rarely.

**Comment by CreativeMD** [comment]
If yes, try to use this version. It will provide more information for me:
https://www.dropbox.com/s/vviewbf3278qlr8/LittleTiles_v1.5.0-pre158_mc1.12.2.jar?dl=1

**Comment by DuncanLevings** [comment]
I was able to reproduce a crash, but not necessarily the same crash error.
Using the same pack and forge version as above with LittleTiles 158.

Steps:
In creative world, I created a small block that i saved as structure and set as fixed axis door with 180 rotation and 20 duration. 
I created a larger door and placed prior structure on face of it. Saved the whole thing as new structure and gave it an advanced animation. 
Move up on y, wait about 20 duration and move back to 0 y. 
I activated the child door and saved. 
Opened and closed the door a few times and it crashed, the amount of time it needs to be opened and closed seems to be random until a crash happens.

> 20:15:12 EDT
> Description: Batching chunks
> 
> java.lang.NullPointerException
>     at com.creativemd.littletiles.common.entity.DoorController.removeWaitingTe(DoorController.java:127)
>     at com.creativemd.littletiles.common.tileentity.TileEntityLittleTiles.clearWaitingAnimations(TileEntityLittleTiles.java:179)
>     at com.creativemd.littletiles.common.tileentity.TileEntityLittleTiles.updateQuadCache(TileEntityLittleTiles.java:206)
>     at com.creativemd.littletiles.client.render.LittleChunkDispatcher.uploadChunk(LittleChunkDispatcher.java:123)
>     at net.minecraft.client.renderer.chunk.ChunkRenderWorker.processTask(SourceFile:126)
>     at net.minecraft.client.renderer.chunk.ChunkRenderWorker.run(SourceFile:43)
>     at java.lang.Thread.run(Unknown Source)


Crash log:
[crash-2019-08-07_20.15.12-client.txt](https://github.com/CreativeMD/LittleTiles/files/3479345/crash-2019-08-07_20.15.12-client.txt)

Another crash but with different error, same steps as above to produce:
[crash-2019-08-07_20.24.37-client.txt](https://github.com/CreativeMD/LittleTiles/files/3479347/crash-2019-08-07_20.24.37-client.txt)

EDIT:
Sometimes I had to spam right click the door while its opening and closing to increase the chance for the crash to happen.
